### PR TITLE
Add getbinpkgonly (-G) to portage

### PR DESCRIPTION
**Pull request by gaima8**
##### SUMMARY
Adds the --getbinpkgonly (-G) option to emerge.

##### ISSUE TYPE
- Feature Pull Request

### CODE REVIEW

File: plugins/modules/packaging/os/portage.py [hunk 1]
```diff
@@ -113,6 +113,12 @@
     choices: [ "web", "yes", "no" ]
     type: str
 
+  getbinpkgonly:
+    description:
+      - Merge only packages specified at PORTAGE_BINHOST in make.conf
+    type: bool
+    default: no
```
Comment (felixfontein) [inline plugins/modules/packaging/os/portage.py:120]:
```suggestion
    default: no
    version_added: 1.3.0
```

File: plugins/modules/packaging/os/portage.py [hunk 2]
```diff
@@ -113,6 +113,12 @@
     choices: [ "web", "yes", "no" ]
     type: str
 
+  getbinpkgonly:
+    description:
+      - Merge only packages specified at PORTAGE_BINHOST in make.conf
```
Comment (felixfontein) [inline plugins/modules/packaging/os/portage.py:118]:
```suggestion
      - Merge only packages specified at C(PORTAGE_BINHOST) in C(make.conf).
```

Comment (felixfontein) [reply plugins/modules/packaging/os/portage.py:118]: Also, you should mention here (and for the other option) that they are mutually exclusive.

**Review by felixfontein — COMMENTED** [review]
You also need a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to).

**Comment by felixfontein** [comment]
(You can ignore the failed ansible/third-party-check, it's unrelated to this PR.)

**Comment by gaima8** [comment]
Requested changes made.
I was a bit hesitant to change the descriptions for `usepkgonly` and `usepkg`, but as I could see no evidence that "This sets getbinpkg=yes." happens for `usepkgonly` I decided to change it.

File: changelogs/fragments/1169-getbinpkgonly.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+  - portage - Add getbinpkgonly option
```
Comment (felixfontein) [inline changelogs/fragments/1169-getbinpkgonly.yaml:3]:
```suggestion
  - portage - add ``getbinpkgonly`` option (https://github.com/ansible-collections/community.general/pull/1169).
```

File: plugins/modules/packaging/os/portage.py [hunk 4]
```diff
@@ -113,21 +113,28 @@
     choices: [ "web", "yes", "no" ]
     type: str
 
+  getbinpkgonly:
+    description:
+      - Merge only packages specified at C(PORTAGE_BINHOST) in C(make.conf). Mutually exclusive with getbinpkg.
```
Comment (felixfontein) [inline plugins/modules/packaging/os/portage.py:118]:
```suggestion
      - Merge only packages specified at C(PORTAGE_BINHOST) in C(make.conf). Mutually exclusive with I(getbinpkg).
```

File: plugins/modules/packaging/os/portage.py [hunk 5]
```diff
@@ -113,21 +113,28 @@
...
+    type: bool
+    default: no
+    version_added: 1.3.0
+
   getbinpkg:
     description:
-      - Prefer packages specified at PORTAGE_BINHOST in make.conf
+      - Prefer packages specified at C(PORTAGE_BINHOST) in C(make.conf). Mutually exclusive with getbinpkgonly.
```
Comment (felixfontein) [inline plugins/modules/packaging/os/portage.py:125]:
```suggestion
      - Prefer packages specified at C(PORTAGE_BINHOST) in C(make.conf). Mutually exclusive with I(getbinpkgonly).
```

File: plugins/modules/packaging/os/portage.py [hunk 6]
```diff
@@ -113,21 +113,28 @@
...
     type: bool
     default: no
 
   usepkgonly:
     description:
-      - Merge only binaries (no compiling). This sets getbinpkg=yes.
+      - Merge only binaries (no compiling). Mutually exclusive with usepkg.
```
Comment (felixfontein) [inline plugins/modules/packaging/os/portage.py:131]:
```suggestion
      - Merge only binaries (no compiling). Mutually exclusive with I(usepkg).
```

File: plugins/modules/packaging/os/portage.py [hunk 7]
```diff
@@ -113,21 +113,28 @@
...
     type: bool
     default: no
 
   usepkg:
     description:
-      - Tries to use the binary package(s) in the locally available packages directory.
+      - Tries to use the binary package(s) in the locally available packages directory. Mutually exclusive with usepkgonly.
```
Comment (felixfontein) [inline plugins/modules/packaging/os/portage.py:137]:
```suggestion
      - Tries to use the binary package(s) in the locally available packages directory. Mutually exclusive with I(usepkgonly).
```

File: changelogs/fragments/1169-getbinpkgonly.yaml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+  - portage - add ``getbinpkgonly`` option (https://github.com/ansible-collections/community.general/pull/1169)
```
Comment (felixfontein) [inline changelogs/fragments/1169-getbinpkgonly.yaml:3]:
```suggestion
  - portage - add ``getbinpkgonly`` option (https://github.com/ansible-collections/community.general/pull/1169).
```

**Comment by wltjr** [comment]
@gaima8 Nice work, thank you for your contributions to furthering the Ansible portage module! These changes seem good and necessary for complete binary package support. However, relating to your comments on changing the the descriptions, this may apply. I could not find any mention of this in the documentation, I literally had to look at the code.

Take a look and see what you think. It seems like it is enabling ```usepkg``` with ```getbinpkg``` and ```usepkgonly``` with ```getbinpkgonly```.
https://github.com/gentoo/portage/blob/5ebc8a249b08318da5a2ca89cee2eed604f7e639/lib/_emerge/actions.py#L2917

Otherwise the changes look good. If the above is valid, it may require some tweaking as to the options. Not sure setting both use and get makes sense, especially if one enables the other.

Thank you again for your efforts and updating your PR to requested changes. That can be frustrating :smile:

**Comment by gaima8** [comment]
Oh, emerge/portage itself is enabling things!
Thanks @wltjr, to me the ansible documentation was saying _ansible_ was doing that.
I don't think it's ansibles job to document internal emerge behaviour though.

emerge also doesn't care how you specify the pkg options.
All at the same time, or in any combinations, is absolutely fine.
Might not do exactly what you *want*, but that isn't ansibles problem.

I've just pushed 734f4d81.

**Comment by wltjr** [comment]
Great! shipit

**Comment by wltjr** [comment]
LGTM

**Comment by felixfontein** [comment]
@gaima8 thanks for fixing this!
@wltjr thanks for reviewing!

**Pull request merged**